docs/mctpd.md

```diff
@@ -3,29 +3,94 @@
+allowing enumeration of managed networks, endpoints and interfaces.
 
-## Bus-owner methods: `au.com.codeconstruct.MCTP.BusOwner1.DRAFT` interface
+## MCTP Interface objects: `/au/com/codeconstruct/interfaces/<name>`
+
+The interface objects represent a connection to a MCTP bus; these will be
+1:1 with the MCTP network interfaces on the system.
+
+### MCTP interface interface: `au.com.codeconstruct.MCTP.Interface1`
+
+All MCTP interface objects host the `au.com.codeconstruct.Interface1` dbus
+interface:
+
+```
+NAME                                 TYPE      SIGNATURE RESULT/VALUE FLAGS
+au.com.CodeConstruct.MCTP.Interface1 interface -         -            -
+.Role                                property  s         "BusOwner"   emits-change writable
+```
+
+The D-Bus interface includes the `Role` property which reports BMC roles
+in the link. The possible value of `Role` are:
 
-This interface exposes bus-owner level functions.
+ * `BusOwner`: this link is the owner of the attached bus,
+ * `Endpoint`: this link is not the owner of the attached bus; and
+ * `Unknown`: not yet configured.
 
-### `.SetupEndpoint`: `say` → `yisb`
+The `Role` property is writable, but it can only be changed when the current
+configured value is `Unknown`. Other platform setup infrastrucut may use
+this to configure the initial MCTP state of the platform.
 
-This method is the normal method used to add a MCTP endpoint. The endpoint is
-identified by MCTP network interface, and physical address. `mctpd` will query
-for the endpoint's current EID, and assign an EID to the endpoint if needed.
-`mctpd` will add local MCTP routes and neighbour table entries for endpoints as
-they are added.
+When the interface `Role` is `BusOwner`, the MCTP interface object will
+also host the `BusOwner1` dbus interface:
 
-`SetupEndpoint <interface name> <hwaddr>`
+### Bus-owner interface: `au.com.codeconstruct.MCTP.BusOwner1` interface
+
+This interface exposes bus-owner level functions, on each interface object that
+represents the bus-owner side of a transport.
+
+```
+NAME                                 TYPE      SIGNATURE RESULT/VALUE FLAGS
+au.com.CodeConstruct.MCTP.Interface1 interface -         -            -
+.Role                                property  s         "BusOwner"   emits-change writable
+au.com.codeconstruct.MCTP.BusOwner1  interface -         -            -
+.AssignEndpoint                      method    ay        yisb         -
+.AssignEndpointStatic                method    ayy       yisb         -
+.LearnEndpoint                       method    ay        yisb         -
+.SetupEndpoint                       method    ay        yisb         -
+```
+
+Those BusOwner methods are:
+
+#### `.SetupEndpoint`: `ay` → `yisb`
+
+This method is the normal method used to add a MCTP endpoint on this interface.
+The endpoint is identified by physical address. `mctpd` will query for the
+endpoint's current EID, and assign an EID to the endpoint if needed. `mctpd`
```

Comment:
`current EID and assigns an EID` I think.

Reply:
no, the "will query" defines the tense there, "assign an EID" is consistent with that.

it would either need to be as it is there, or 

    `mctpd` queries for the endpoint's current EID, and assigns an EID to the endpoint if needed

but I find that wording a little too passive.